yunwoo/programmers-1/체육복.js

```diff
@@ -0,0 +1,27 @@
+function setSortAndFliter(arr, arr2) {
+  return arr.sort((a, b) => a - b).filter((v) => !arr2.includes(v));
+}
+
+function solution(n, lost, reserve) {
+  const noPeClothesStudent = setSortAndFliter(lost, reserve);
+  const extraPeClothesStudent = setSortAndFliter(reserve, lost);
+
+  for (const el of noPeClothesStudent) {
+    const extraUniformidx = extraPeClothesStudent.findIndex(
+      (value) => value === el - 1 || value === el + 1
+    );
+
+    extraUniformidx !== -1
+      ? extraPeClothesStudent.splice(extraUniformidx, 1)
+      : n--;
+  }
+
+  return n;
+}
```

Comment:
```suggestion
function setSortAndFliter(arr, arr2) {
  return arr.sort((a, b) => a - b).filter((v) => !arr2.includes(v));
}

function solution(n, lost, reserve) {
  const noPeClothesStudent = setSortAndFliter(lost, reserve);
  const extraPeClothesStudent = setSortAndFliter(reserve, lost);

  for (const el of noPeClothesStudent) {
    const extraUniformIdx = extraPeClothesStudent.findIndex(
      (value) => value === el - 1 || value === el + 1
    );

    extraUniformIdx !== -1
      ? extraPeClothesStudent.splice(extraUniformIdx, 1)
      : n--;
  }

  return n;
}
```
- `findIndex`에서 조건식을 prev를 앞에 둬서 먼저 prev를 구하고 없으면 next를 찾음으로써, prev, next 각각 index구하지 않고 하나의 findIndex 메서드로 찾은거 너무 좋은 풀이네요!! 배웠습니다 👍
- 궁금한데 noPe~~에서 `Pe`가 의미일까요? 😂 `Uniform`과 `Clothes`가 하나로 단어를 통일해주면 어땠을까 싶습니다 :) 여기서는 같은 의미로 사용되는 것 같아서요 ㅎㅎ
- 또한, `extraUniformidx` 를 카멜 케이스로 수정하였습니다 ~!
- 그외에는 잘푸셨습니다 고생하셨습니다 👍

Reply:
- 체육복을 구글링 해봤는데 `PeClothes` 이 체육복이라 써있었는데 아니였나 보군요..!🥲
- 컨벤션 조금 더 신경 쓰겠습니다!!! 리뷰 감사합니다🙇‍♂️🙇‍♂️👍!!